pkg/stub/handler.go

```diff
@@ -1,102 +1,119 @@
 package stub
 
 import (
-	"context"
+	"time"
 
 	"github.com/aerogear/mobile-developer-console/pkg/apis/aerogear/v1alpha1"
 	"github.com/aerogear/mobile-developer-console/pkg/mobile"
-	"github.com/operator-framework/operator-sdk/pkg/sdk"
-	log "github.com/sirupsen/logrus"
+	"github.com/labstack/gommon/log"
 	"k8s.io/api/core/v1"
 )
 
-func NewHandler(mobileAppRepo mobile.MobileClientRepo) sdk.Handler {
-	return &Handler{
-		mobileClientRepo: mobileAppRepo,
+func WatchSecrets(namespace string, secretsCRUDL mobile.SecretsCRUDL, mobileAppRepo mobile.MobileClientRepo) {
+	getWatchInterface := secretsCRUDL.Watch(namespace)
+	watchInterface, err := getWatchInterface()
+	if err != nil {
+		return
 	}
-}
-
-type Handler struct {
-	// Fill me
-	mobileClientRepo mobile.MobileClientRepo
-}
-
-func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	switch o := event.Object.(type) {
-	case *v1.Secret:
-		secret := o
-
-		if !isValidSecret(secret) {
-			return nil
+	events := watchInterface.ResultChan()
+	defer func() {
+		if watchInterface != nil {
+			watchInterface.Stop()
 		}
-
-		labels := secret.GetLabels()
-		clientId := labels["clientId"]
-
-		app, err := h.mobileClientRepo.ReadByName(clientId)
-		if err != nil {
-			log.Errorf("failed to read mobile client by name %v", clientId)
-			return err
+	}()
+	willHandle := false
+	for {
+		select {
+		case _, ok := <-events:
```

Comment:
from my testing, this is getting triggered many times on server start. I think it is because the API returned all the events from the beginning. Maybe we should update the watch option so that it will only return the latest events?